### DISCUSSION TITLE
ld: cannot find crti.o: No such file or directory

### DISCUSSION THREAD

**Discussion by jhnworks** [Homebrew/discussions#1984] [Linux]
```
[@~]$ whereis gcc
gcc: /usr/bin/gcc /usr/lib/gcc /usr/libexec/gcc /usr/share/man/man1/gcc.1.gz
[@~]$ gcc --version
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

/tmp/gccA5-20210815-9439-1tsar7e/gcc-5.5.0/build/./gcc/xgcc -B/tmp/gccA5-20210815-9439-1tsar7e/gcc-5.5.0/build/./gcc/ -B/home/B1151/.linuxbrew/Cellar/gcc@5/5.5.0_7/x86_64-unknown-linux-gnu/bin/ -B/home/B1151/.linuxbrew/Cellar/gcc@5/5.5.0_7/x86_64-unknown-linux-gnu/lib/ -isystem /home/B1151/.linuxbrew/Cellar/gcc@5/5.5.0_7/x86_64-unknown-linux-gnu/include -isystem /home/B1151/.linuxbrew/Cellar/gcc@5/5.5.0_7/x86_64-unknown-linux-gnu/sys-include    -O2  -g -O2 -DIN_GCC    -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-format -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition  -isystem ./include   -fpic -mlong-double-80 -g -DIN_LIBGCC2 -fbuilding-libgcc -fno-stack-protector  -shared -nodefaultlibs -Wl,--soname=libgcc_s.so.1 -Wl,--version-script=libgcc.map -o ./libgcc_s.so.1.tmp -g -O2 -B./ _muldi3_s.o _negdi2_s.o _lshrdi3_s.o _ashldi3_s.o _ashrdi3_s.o _cmpdi2_s.o _ucmpdi2_s.o _clear_cache_s.o _trampoline_s.o __main_s.o _absvsi2_s.o _absvdi2_s.o _addvsi3_s.o _addvdi3_s.o _subvsi3_s.o _subvdi3_s.o _mulvsi3_s.o _mulvdi3_s.o _negvsi2_s.o _negvdi2_s.o _ctors_s.o _ffssi2_s.o _ffsdi2_s.o _clz_s.o _clzsi2_s.o _clzdi2_s.o _ctzsi2_s.o _ctzdi2_s.o _popcount_tab_s.o _popcountsi2_s.o _popcountdi2_s.o _paritysi2_s.o _paritydi2_s.o _powisf2_s.o _powidf2_s.o _powixf2_s.o _mulsc3_s.o _muldc3_s.o _mulxc3_s.o _divsc3_s.o _divdc3_s.o _divxc3_s.o _bswapsi2_s.o _bswapdi2_s.o _clrsbsi2_s.o _clrsbdi2_s.o _fixunssfsi_s.o _fixunsdfsi_s.o _fixunsxfsi_s.o _fixsfdi_s.o _fixdfdi_s.o _fixxfdi_s.o _fixunssfdi_s.o _fixunsdfdi_s.o _fixunsxfdi_s.o _floatdisf_s.o _floatdidf_s.o _floatdixf_s.o _floatundisf_s.o _floatundidf_s.o _floatundixf_s.o _divdi3_s.o _moddi3_s.o _udivdi3_s.o _umoddi3_s.o _udiv_w_sdiv_s.o _udivmoddi4_s.o cpuinfo_s.o sfp-exceptions_s.o addtf3_s.o divtf3_s.o multf3_s.o negtf2_s.o subtf3_s.o unordtf2_s.o fixtfsi_s.o fixunstfsi_s.o floatsitf_s.o floatunsitf_s.o fixtfdi_s.o fixunstfdi_s.o floatditf_s.o floatunditf_s.o fixtfti_s.o fixunstfti_s.o floattitf_s.o floatuntitf_s.o extendsftf2_s.o extenddftf2_s.o extendxftf2_s.o trunctfsf2_s.o trunctfdf2_s.o trunctfxf2_s.o getf2_s.o letf2_s.o eqtf2_s.o _divtc3_s.o _multc3_s.o _powitf2_s.o enable-execute-stack_s.o unwind-dw2_s.o unwind-dw2-fde-dip_s.o unwind-sjlj_s.o unwind-c_s.o emutls_s.o libgcc.a -lc && rm -f ./libgcc_s.so && if [ -f ./libgcc_s.so.1 ]; then mv -f ./libgcc_s.so.1 ./libgcc_s.so.1.backup; else true; fi && mv ./libgcc_s.so.1.tmp ./libgcc_s.so.1 && ln -s libgcc_s.so.1 ./libgcc_s.so
ld: cannot find crti.o: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [libgcc_s.so] Error 1
make[3]: Leaving directory `/tmp/gccA5-20210815-9439-1tsar7e/gcc-5.5.0/build/x86_64-unknown-linux-gnu/libgcc'
make[2]: *** [all-stage1-target-libgcc] Error 2
make[2]: Leaving directory `/tmp/gccA5-20210815-9439-1tsar7e/gcc-5.5.0/build'
make[1]: *** [stage1-bubble] Error 2
make[1]: Leaving directory `/tmp/gccA5-20210815-9439-1tsar7e/gcc-5.5.0/build'
make: *** [bootstrap] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
`brew install gcc` requires gcc https://github.com/Homebrew/linuxbrew-core/issues/23436
gcc@5: force cellar: :any on Linux https://github.com/Homebrew/linuxbrew-core/pull/23361
`ld: cannot find crti.o` error when installing `gcc@5` on CentOS 7 https://github.com/Homebrew/linuxbrew-core/issues/23460
```
How does one get around this?

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/linuxbrew-core/issues/23460, which is linked in the output, seems to have a solution

**Reply by jhnworks** [reply]
Everything results in this..

`

ld: cannot find crti.o: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [libgcc_s.so] Error 1
make[3]: Leaving directory `/tmp/gccA5-20210815-332-wxg68/gcc-5.5.0/build/x86_64-unknown-linux-gnu/libgcc'
make[2]: *** [all-stage1-target-libgcc] Error 2
make[2]: Leaving directory `/tmp/gccA5-20210815-332-wxg68/gcc-5.5.0/build'
make[1]: *** [stage1-bubble] Error 2
make[1]: Leaving directory `/tmp/gccA5-20210815-332-wxg68/gcc-5.5.0/build'
make: *** [bootstrap] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
`brew install gcc` requires gcc https://github.com/Homebrew/linuxbrew-core/issues/23436
gcc@5: force cellar: :any on Linux https://github.com/Homebrew/linuxbrew-core/pull/23361
`ld: cannot find crti.o` error when installing `gcc@5` on CentOS 7 https://github.com/Homebrew/linuxbrew-core/issues/23460
[user1@ ~]$
[user1@ ~]$ brew install --force-bottle gcc@5
Warning: HOMEBREW_NO_ENV_FILTERING is undocumented, deprecated and will be removed in a future Homebrew release (because it breaks many things)!
Warning: HOMEBREW_NO_ENV_FILTERING is undocumented, deprecated and will be removed in a future Homebrew release (because it breaks many things)!
Warning: HOMEBREW_NO_ENV_FILTERING is undocumented, deprecated and will be removed in a future Homebrew release (because it breaks many things)!
Error: undefined method `fetch' for nil:NilClass
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/formula_installer.rb:1133:in `fetch'
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/install.rb:299:in `install_formula'
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:207:in `block in install'
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:205:in `each'
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:205:in `install'
/home/user1/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:129:in `<main>'


`

**Reply by SMillerDev** [reply]
> Warning: HOMEBREW_NO_ENV_FILTERING is undocumented, deprecated and will be removed in a future Homebrew release (because it breaks many things)!

This is definitely a warning that shouldn't be ignored, try unsetting `HOMEBREW_NO_ENV_FILTERING`

**Reply by jhnworks** [reply]
Still no dice.

**Reply by danielnachun** [reply]
You have not changed the cellar for `gcc@5` to be `:any` for Linux.  You need to use `brew edit gcc@5` as the instructions say to do here: https://github.com/Homebrew/linuxbrew-core/issues/23460#issuecomment-859805249.  The `gcc` formulae don't build from source right now when using brewed `glibc`, and I'm hoping we'll soon have a solution so that those formula are given the proper `cellar :any` they need on Linux, so that you don't have to build them from source anymore.
